tests/operators/test_agnostic_load_file.py

```diff
@@ -466,47 +497,84 @@ def test_aql_gcs_file_to_postgres(self):
-            warehouse=os.getenv("SNOWFLAKE_WAREHOUSE"),
-        )
-
-        # Drop target table
-        hook.run(
-            f"DROP TABLE IF EXISTS {os.getenv('SNOWFLAKE_SCHEMA')}.{self.SNOWFLAKE_OUTPUT_TABLE_NAME}"
-        )
-        self.create_and_run_task(
-            load_file,
-            (),
-            {
-                "path": str(self.cwd) + "/../data/homes.csv",
-                "file_conn_id": "",
-                "output_table": Table(
-                    table_name=self.SNOWFLAKE_OUTPUT_TABLE_NAME,
-                    database=os.getenv("SNOWFLAKE_DATABASE"),
-                    schema=os.getenv("SNOWFLAKE_SCHEMA"),
-                    warehouse=os.getenv("SNOWFLAKE_WAREHOUSE"),
-                    conn_id="snowflake_conn",
-                ),
-            },
-        )
-
-        # Read table from db
-        df = hook.get_pandas_df(
-            f"SELECT * FROM {os.getenv('SNOWFLAKE_SCHEMA')}.{self.SNOWFLAKE_OUTPUT_TABLE_NAME}"
-        )
 
-        assert df.iloc[0].to_dict() == {
-            "SELL": 142.0,
-            "LIST": 160.0,
-            "LIVING": 28.0,
-            "ROOMS": 10.0,
-            "BEDS": 5.0,
-            "BATHS": 3.0,
-            "AGE": 60.0,
-            "ACRES": 0.28,
-            "TAXES": 3167.0,
-        }
+@pytest.fixture
+def sample_dag():
+    yield DAG("test_dag", default_args={"owner": "airflow", "start_date": DEFAULT_DATE})
+    with create_session() as session:
+        session.query(DagRun).delete()
+        session.query(TI).delete()
+
+
+def create_and_run_task(dag, decorator_func, op_args, op_kwargs):
+    with dag:
+        function = decorator_func(*op_args, **op_kwargs)
+
+    _ = dag.create_dagrun(
+        run_id=DagRunType.MANUAL.value,
+        start_date=timezone.utcnow(),
+        data_interval=[DEFAULT_DATE, DEFAULT_DATE],
+        execution_date=DEFAULT_DATE,
+        state=State.RUNNING,
+    )
+    function.operator.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE)
+    return function
+
+
+@pytest.fixture
+def sql_server(request):
+    sql_name = request.param
+    hook_parameters = SQL_SERVER_HOOK_PARAMETERS.get(sql_name)
+    hook_class = SQL_SERVER_HOOK_CLASS.get(sql_name)
+    if hook_parameters is None or hook_class is None:
+        raise ValueError(f"Unsupported SQL server {sql_name}")
+    hook = hook_class(**hook_parameters)
+    schema = hook_parameters.get("schema", DEFAULT_SCHEMA)
+    hook.run(f"DROP TABLE IF EXISTS {schema}.{OUTPUT_TABLE_NAME}")
+    yield (sql_name, hook)
+    hook.run(f"DROP TABLE IF EXISTS {schema}.{OUTPUT_TABLE_NAME}")
+
+
+@mock.patch.dict(
+    os.environ,
+    {
+        "AIRFLOW__ASTRO__CONN_AWS_DEFAULT": "abcd:%40%23%24%25%40%24%23ASDH%40Ksd23%25SD546@"
+    },
+)
+def test_aws_decode():
+    from astro.utils.cloud_storage_creds import parse_s3_env_var
+
+    k, v = parse_s3_env_var()
+    assert v == "@#$%@$#ASDH@Ksd23%SD546"
+
+
+@pytest.mark.parametrize("sql_server", ["snowflake", "postgres"], indirect=True)
```

Comment:
@utkarsharma2 what do you think of adding BQ to this test?

Reply:
I think that sounds like a good idea.